### PR TITLE
Deployment version

### PR DESCRIPTION
**Pull request by cicdw**
### Summary

Wires up deployment versions to the client on the heels of #6170

### CODE REVIEW

**Comment by anna-geller** [comment]
Works like a charm!

image

image

**Comment by anna-geller** [comment]
@cicdw should we loop in Nicholas to add this field to the UI as well? would be really nice for troubleshooting - imagine you can immediately check from the UI the version indicating your Git commit and then check the corresponding commit in your repo when something goes wrong

**Review by anna-geller — APPROVED** [review]

File: src/prefect/deployments.py [hunk 1]
```diff
@@ -1,51 +1,5 @@
 """
 Objects for specifying deployments and utilities for loading flows from deployments.
-
```
Comment (chrisguidry) [inline src/prefect/deployments.py:3]: Yup

**Review by chrisguidry — APPROVED** [review]
Looks good to me if we are only considering `version` as an additional piece of user-provided metadata on the `Flow`, and not changing the cardinality of `Flows`.

I do wonder a bit about the user expectation.  Would people expect there to be multiple `Flows` at a time, with different versions?  That definitely blows up the scope of this change, so we probably shouldn't go there yet.  Just something to think about, we may want to have some explanation in the docs about `version`.

I think we could make `version` more valuable without a lot of effort if we add `flow_version` to `FlowRun`, so when you look back at past runs, you see which version of the flow was running at that time.

File: src/prefect/deployments.py (hunk 1 above)
Comment (anna-geller) [reply src/prefect/deployments.py:3]: @chrisguidry I would expect that after adding a version on a deployment, users might leverage that for versioning their deployed flows (e.g. with the version being GITHUB_SHA) and will use the version set on the flow decorator only to version flows triggered without deployments (e.g., triggered locally or e.g. from Amazon Lambda function)

afaik, we currently don't expose this version anywhere in the UI or CLI, so currently, we only use this attribute internally without exposing it in any way to users

**Comment by cicdw** [comment]
@chrisguidry the `version` provided on the flow decorator is already being used to populate a `version` field on the flow run table 👍

**Review by madkinsz — APPROVED** [review]

**Comment by chrisguidry** [comment]
> @chrisguidry the `version` provided on the flow decorator is already being used to populate a `version` field on the flow run table 👍 

Ohhhh, thank you, that sounds good

**Pull request merged**